collections/soldiers.js

```diff
@@ -0,0 +1,34 @@
+const MongoClient = require('mongodb').MongoClient;
+
+const url = 'mongodb://localhost:27017';
+const dbName = 'myproject';
+
+function checkData(soldierData) {
+  if (/tt[0-9]{7}/.test(soldierData["id"]) &&
+    soldierData["name"] &&
+    soldierData["rank"] &&
+    soldierData["limitations"] &&
+    Object.keys(soldierData).length === 5) {
+    return true;
+  }
+  return false;
+}
+
+function insertSoldier(soldierData, done) {
+  soldierData["duties"] = [];
```

Comment:
`soldierData["duties"] = soldierData["duties"] || [];`
You can't accidentally erase data that the user sent for you to save.
If they shouldn't post with duties in this phase - block them from doing so!